Automattic/wp-calypso
# Domains: Implement transfer-in precheck/options screens

As we're introducing incoming domain transfers, we have a precheck screen, to confirm that everything is ok with the domain, and an options screen for the transfer itself (nameservers, importing DNS, privacy). These appear before the product is added to the cart, and the last screen leads to checkout.

- Move screen out of checkout
- Show domain details form in checkout if transfer in present
- Style the screens
- Submit the transfer product with extra info

Test:
- Domains -> Add
- Use a Domain I Own
- Input a domain you own. Make sure it takes you to the precheck screen.
- Unlock your domain, click Check Again. Make sure the status is properly picked.
- Check the e-mail shown as the admin email.
- Proceed to Options screen
- Make sure the options work and are submitted properly.
- Make sure the price is ok.
- Initiate the transfer.
- Check your e-mail for confirmation.

Requires on: D8184-code
